src/cobra/flux_analysis/deletion.py

```diff
@@ -405,3 +401,60 @@ def double_gene_deletion(
         processes=processes,
         **kwargs
     )
+
+
+@pd.api.extensions.register_dataframe_accessor("knockout")
+class KnockoutAccessor:
+    """Access unique combinations of reactions in deletion results."""
+
+    def __init__(self, pandas_obj):
+        """Set up the accessor.
+
+        Parameters:
+        -----------
+        pandas_obj : pd.DataFrame or pd.Series
+            A result from one of the deletion methods.
+        """
+        self._validate(pandas_obj)
+        self._result = pandas_obj
+
+    @staticmethod
+    def _validate(obj):
+        # verify it is a deletion results
+        if any(name not in obj.columns for name in ["ids", "growth", "status"]):
+            raise AttributeError("Must be DataFrame returned by a deletion method.")
+
+    def __getitem__(self, args):
+        """Return the deletion result for a particular set of knocked entities.
+
+        Parameters:
+        -----------
+        args : iterable of str, cobra.Reaction or cobra.Gene
+
+        Returns:
+        --------
+        pd.DataFrame
+            The deletion result where the chosen entities have been deleted.
+        """
+        if not any(isinstance(args, t) for t in [tuple, list]):
+            args = [args]
+
+        if any(isinstance(args[0], t) for t in [Reaction, Gene, str]):
```

Comment:
Yeah, so my motivation was to have consistent indexed n behavior for single and double deletions. So that pads ng two elements to the indexed always returns two rows. I felt it would be careful infusing otherwise, but we could do it the other way. Would just be a bit more code since you now have to treat single deletions differently from multi-deletions.

Reply:
Let's keep it that way then.